TDSClient/TDSClient/TDS/Client/TDSSQLTestClient.cs

```diff
@@ -0,0 +1,223 @@
+    {
+        public string Server { get; private set; }
+
+        public string ServerName { get; private set; }
+
+        public int Port { get; private set; }
+
+        public string UserID { get; private set; }
+
+        public string Password { get; private set; }
+
+        public string Database { get; private set; }
+
+        public TDSCommunicator TdsCommunicator { get; private set; }
+        public TcpClient Client { get; private set; }
+        public TDSClientVersion Version { get; private set; }
+
+        public SslProtocols EncryptionProtocol { get; private set; }
+
+        private bool Reconnect;
+
+        public TDSSQLTestClient(string server, int port, string userID, string password, string database, SslProtocols encryptionProtocol = SslProtocols.Tls12)
+        {
+            if (string.IsNullOrEmpty(server) || string.IsNullOrEmpty(userID) || string.IsNullOrEmpty(password) || string.IsNullOrEmpty(database))
+            {
+                throw new ArgumentNullException();
+            }
+
+            Client = null;
+            Version = new TDSClientVersion(1, 0, 0, 0);
+            Server = server;
+            ServerName = server;
+            Port = port;
+            UserID = userID;
+            Password = password;
+            Database = database;
+            EncryptionProtocol = encryptionProtocol;
+
+            LoggingUtilities.WriteLog($" Instantiating TDSSQLTestClient with the following parameters:");
+
+            LoggingUtilities.WriteLog($"     Server: {server}.");
+            LoggingUtilities.WriteLog($"     Port: {port}.");
+            LoggingUtilities.WriteLog($"     UserID: {userID}.");
+            LoggingUtilities.WriteLog($"     Database: {database}.");
+        }
+
+        public void SendPreLogin()
+        {
+            LoggingUtilities.WriteLog($" SendPreLogin initiated.");
+            var tdsMessageBody = new TDSPreLoginPacketData(Version);
+
+            tdsMessageBody.AddOption(TDSPreLoginOptionTokenType.Encryption, TDSEncryptionOption.EncryptOff);
+            tdsMessageBody.Terminate();
+
+            TdsCommunicator.SendTDSMessage(tdsMessageBody);
+            LoggingUtilities.WriteLog($" SendPreLogin done.");
+        }
+
+        public void SendLogin7()
+        {
+            LoggingUtilities.WriteLog($" SendLogin7 initiated.");
+
+            var tdsMessageBody = new TDSLogin7PacketData();
+
+            tdsMessageBody.AddOption("HostName", (ushort)Environment.MachineName.Length, Environment.MachineName);
+            tdsMessageBody.AddOption("UserName", (ushort)UserID.Length, UserID);
+            tdsMessageBody.AddOption("ServerName", (ushort)ServerName.Length, ServerName);
+            tdsMessageBody.AddOption("Password", (ushort)Password.Length, Password);
+            tdsMessageBody.AddOption("Database", (ushort)Database.Length, Database);
+            tdsMessageBody.AddOption("IntName", (ushort)"TDSSQLTestClient".Length, "TDSSQLTestClient");
```

Comment:
Create "TDSSQLTestClient" as a constant, its already repeated twice in this line.